src/management/projects.ts

```diff
@@ -36,6 +40,52 @@ export interface ProjectParentResource {
   type: ProjectParentResourceType;
 }
 
+export const PROJECTS_CREATE_QUESTIONS: Question[] = [
+  {
+    type: "input",
+    name: "projectId",
+    default: "",
+    message:
+      "Please specify a unique project id " +
+      `(${clc.yellow("warning")}: cannot be modified afterward) [6-30 characters]:\n`,
+  },
+  {
+    type: "input",
+    name: "displayName",
+    default: "",
+    message: "What would you like to call your project? (defaults to your project ID)",
+  },
+];
+
+export async function createFirebaseProject(
+  projectId: string,
+  options: { displayName?: string; parentResource?: ProjectParentResource }
+): Promise<FirebaseProjectMetadata> {
```

Comment:
@mbleigh, @bkendall, @nbegley, we agreed before that API methods should not contain output (logger, spinner) behaviors. But since we are reusing the flow for the `firebase:init`, I think that moving `createFirebaseProject()` into `projects.ts` is necessary.

This also enables testing spinner behavior, but I need your approvals before doing so

Reply:
FWIW my objection was never to the idea of "interactive / spinner stuff can never be in anything other than the command itself" it's that we shouldn't be mixing API utility methods with interactive user prompting and output.

I'm fine with this moving to `projects.ts` but let's call it something like `interactivelyCreateProject` so it's clear this will trigger user interaction and things.

I think the rule of thumb should be simply that we should not mix together terminal UI code with direct API calls (e.g. `api.request`).

Seem reasonable @bkendall?

Reply:
@TrCaM please rename `createNewProject` to `promptAndCreateProject` and `createFirebaseProject` to `createFirebaseProjectAndLog`.

The names are a bit obtuse, but it will prevent anyone from accidentally using those methods in the future thinking they're just API utilities.

Reply:
SGTM, thanks